dbcon/execplan/treenode.h

```diff
@@ -1127,6 +1127,17 @@ inline int64_t TreeNode::getDatetimeIntVal()
     else if (fResultType.colDataType == execplan::CalpontSystemCatalog::DATETIME)
         //return (fResult.intVal & 0xFFFFFFFFFFF00000LL);
         return (fResult.intVal);
+    else if (fResultType.colDataType == execplan::CalpontSystemCatalog::TIMESTAMP)
+    {
+        dataconvert::TimeStamp timestamp(fResult.intVal);
+        int64_t seconds = timestamp.second;
+        dataconvert::MySQLTime m_time;
+        dataconvert::gmtSecToMySQLTime(seconds, m_time, timeZone);
+        dataconvert::DateTime dt(m_time.year, m_time.month, m_time.day, m_time.hour,
+                                 m_time.minute, m_time.second, timestamp.msecond);
+        memcpy(&fResult.intVal, &dt, 8);
```

Comment:
`DateTime`, `Time` and `TimeStamp` are 64-bit structs, we cannot add a member variable to them.

Reply:
Very good. Then Result should have a setter to assign a int value. And this is a very good example why I would like to see setters/getters used instead of direct attributes manipulations. Let me explain. You are setting Result::intVal. Imagine this Result needs to be used as an unsigned int(I am aware about TreeNode used in such context but for the sake of the explanation I will ignore this). Some Result::uIntValue will be used instead. And this potentially gives us hard to detect issues. With a setter you can set a value properly though.